tests/networkpolicy_test.go

```diff
@@ -1,173 +1,234 @@
 
-	Context("vms limited by deny by label networkpolicy", func() {
-		var policy *v1network.NetworkPolicy
-
-		BeforeEach(func() {
-			// deny-by-label networkpolicy will deny the traffice for the vm which have the same label
-			By("Create deny-by-label networkpolicy")
-			networkpolicy := &v1network.NetworkPolicy{
-				ObjectMeta: v13.ObjectMeta{
-					Name: "deny-by-label",
-				},
-				Spec: v1network.NetworkPolicySpec{
-					PodSelector: v13.LabelSelector{
-						MatchLabels: map[string]string{
-							"type": "test",
+			BeforeEach(func() {
+				// deny-by-label networkpolicy will deny the traffice for the vm which have the same label
+				By("Create deny-by-label networkpolicy")
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "deny-by-label",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"type": "test",
+							},
 						},
+						Ingress: []networkv1.NetworkPolicyIngressRule{},
 					},
-					Ingress: []v1network.NetworkPolicyIngressRule{},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
-		})
+				}
+				policy = createNetworkPolicy(serverVMI.Namespace, "deny-by-label", metav1.LabelSelector{MatchLabels: serverVMILabels}, []networkv1.NetworkPolicyIngressRule{})
+			})
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
 
-		It("[test_id:1515] should be failed to reach vmia from vmic", func() {
-			By("Connect vmia from vmic")
-			assertPingFail(vmic, vmia)
-		})
+			It("[test_id:1515] should be failed to reach serverVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect serverVMI from clientVMIAlternativeNamespace")
+				assertPingFail(clientVMIAlternativeNamespace, serverVMI)
+			})
 
-		It("[test_id:1516] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
-		})
+			It("[test_id:1516] should be failed to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertPingFail(clientVMI, serverVMI)
+			})
+
+			It("[test_id:1517] should be successful to reach clientVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect clientVMI from clientVMIAlternativeNamespace")
+				assertPingSucceed(clientVMIAlternativeNamespace, clientVMI)
+			})
 
-		It("[test_id:1517] should be successful to reach vmib from vmic", func() {
-			By("Connect vmib from vmic")
-			assertPingSucceed(vmic, vmib)
+		})
+		Context("and an allow all http ports policy is populated", func() {
+			var policy *networkv1.NetworkPolicy
+			BeforeEach(func() {
+				port80 := intstr.FromInt(80)
+				port81 := intstr.FromInt(81)
+				tcp := corev1.ProtocolTCP
```

Comment:
If you use this more than once, consider making it a scoped constant (or even global in this file).